solana-programs/anchor/audius-data/programs/audius-data/src/utils.rs

```diff
@@ -7,13 +7,14 @@ use anchor_lang::{prelude::*, solana_program::system_program};
 pub fn validate_user_authority<'info>(
     program_id: &Pubkey,
     user: &Account<'info, User>,
-    user_delegate_authority: &AccountInfo<'info>,
+    user_authority_delegate: &AccountInfo<'info>,
     authority: &Signer,
+    authority_delegation_status: &AccountInfo<'info>,
 ) -> Result<()> {
     if user.authority != authority.key() {
-        // Reject if system program provided as user delegate_auth
-        if user_delegate_authority.key() == system_program::id() {
-            return Err(ErrorCode::Unauthorized.into());
+        // Reject if system program provided as user_authority_delegate
+        if user_authority_delegate.key() == system_program::id() {
```

Comment:
should this be an OR condition on authority_delegation_status? or was that intentionally left out

Reply:
oh yeah you're right, fixing now